homeassistant/components/konnected.py

```diff
@@ -0,0 +1,298 @@
+
+            actuator_status = next((actuator for actuator in
+                                    self.status.get('actuators') if
+                                    actuator.get('pin') == pin), {})
+            if actuator_status.get(ATTR_STATE):
+                initial_state = bool(int(actuator_status.get(ATTR_STATE)))
+            else:
+                initial_state = None
+
+            actuators[pin] = {
+                CONF_NAME: entity.get(
+                    CONF_NAME, 'Konnected {} Actuator {}'.format(
+                        self.device_id[6:], PIN_TO_ZONE[pin])),
+                ATTR_STATE: initial_state,
+                'activation': entity['activation'],
+            }
+            _LOGGER.info('Set up actuator %s (initial state: %s)',
+                         actuators[pin].get(CONF_NAME),
+                         actuators[pin].get(ATTR_STATE))
+
+        device_data = {
+            'client': self.client,
+            CONF_SENSORS: sensors,
+            CONF_SWITCHES: actuators,
+            CONF_HOST: self.host,
+            CONF_PORT: self.port,
+        }
+
+        if 'devices' not in self.hass.data[DOMAIN]:
+            self.hass.data[DOMAIN][CONF_DEVICES] = {}
+
+        _LOGGER.info('Storing data in hass.data[konnected]: %s', device_data)
+        self.hass.data[DOMAIN][CONF_DEVICES][self.device_id] = device_data
+
+    @property
+    def stored_configuration(self):
+        """Return the configuration stored in `hass.data` for this device."""
+        return self.hass.data[DOMAIN][CONF_DEVICES][self.device_id]
+
+    def sensor_configuration(self):
+        """Return the configuration map for syncing sensors."""
+        return [{'pin': p} for p in
+                self.stored_configuration[CONF_SENSORS].keys()]
+
+    def actuator_configuration(self):
+        """Return the configuration map for syncing actuators."""
+        return [{'pin': p,
+                 'trigger': (0 if data.get('activation') in [0, 'low'] else 1)}
+                for p, data in
+                self.stored_configuration[CONF_SWITCHES].items()]
+
+    def sync_device(self):
+        """Sync the new pin configuration to the Konnected device."""
+        desired_sensor_configuration = self.sensor_configuration()
+        current_sensor_configuration = [
+            {'pin': s[CONF_PIN]} for s in self.status.get('sensors')]
+        _LOGGER.info('%s: desired sensor config: %s', self.device_id,
+                     desired_sensor_configuration)
+        _LOGGER.info('%s: current sensor config: %s', self.device_id,
+                     current_sensor_configuration)
+
+        desired_actuator_config = self.actuator_configuration()
+        current_actuator_config = self.status.get('actuators')
+        _LOGGER.info('%s: desired actuator config: %s', self.device_id,
+                     desired_actuator_config)
+        _LOGGER.info('%s: current actuator config: %s', self.device_id,
+                     current_actuator_config)
```

Comment:
These log messages should IMO not be `info`. Home Assistant's default is to print every log message above and including `info`. If someone needs these log messages they can still manually set the log level for this component using the [logger component](https://www.home-assistant.io/components/logger/).

Reply:
You're right, these were really meant for debugging discovery. I will change them to `debug`.